### PR TITLE
Fix VS Warning

### PR DESCRIPTION
**Pull request by wookieejedi**
If this is trivial and not needed feel free to dismiss, but this one-line PR simply fixes a warning that keeps showing up when building with VS.

### CODE REVIEW

File: code/ship/ship.cpp [hunk 1]
```diff
@@ -10087,7 +10087,7 @@ int ship_create(matrix* orient, vec3d* pos, int ship_type, const char* ship_name
 		strcpy_s(shipp->ship_name, base_name);
 
 		// if generated name will be longer than allowable name, truncate the class section of the name by the overflow
-		int char_overflow = (strlen(base_name) + strlen(suffix)) - (NAME_LENGTH - 1);
+		int char_overflow = static_cast <int> (strlen(base_name) + strlen(suffix)) - (NAME_LENGTH - 1);
```
Comment (JohnAFernandez) [inline code/ship/ship.cpp:10090]: I think it's a fine quick change to make. But static cast the individual strlen calls. m!m mentioned on a review in the past that putting more than one thing in a cast expression doesn't work as you would expect.

**Review by JohnAFernandez — DISMISSED** [review]

File: code/ship/ship.cpp (hunk 1 above)
Comment (wookieejedi) [reply code/ship/ship.cpp]: Thanks for the heads up, that is excellent to know. I've updated so each is cast individually.

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Let's make sure this is trhe appropriate fix.  What is the Visual Studio warning?

**Comment by wookieejedi** [comment]
Good question, `'initializing': conversion from 'size_t' to 'int', possible loss of data`

**Review by Goober5000 — CHANGES_REQUESTED** [review]
~~In that case, try removing the existing casts, but casting `char_overflow` to `size_t`.~~

**Comment by Goober5000** [comment]
Try this instead:
```
int char_overflow = static_cast<int>(strlen(base_name) + strlen(suffix)) - (NAME_LENGTH - 1);
if (char_overflow > 0) {
	shipp->ship_name[strlen(base_name) - static_cast<size_t>(char_overflow)] = '\0';
}
```

**Review by Goober5000 — APPROVED** [review]

**Pull request merged**